ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpImplTreeBuilder.fs

```diff
@@ -1106,8 +1109,10 @@ type FSharpExpressionTreeBuilder(lexer, document, lifetime, path, projectedOffse
 
         | SynExpr.DebugPoint _ -> failwithf $"Synthetic expression: {expr}"
 
-    member x.ProcessAndLocalBinding(pat: SynPat, expr: SynExpr) =
-        x.PushRangeForMark(expr.Range, x.Mark(pat.Range), ElementType.LOCAL_BINDING)
+    member x.ProcessAndLocalBinding(pat: SynPat, expr: SynExpr, range) =
+        x.AdvanceToTokenOrRangeStart(FSharpTokenType.AND_BANG, range)
```

Comment:
Why does it only look for `and!` token? Is `and` handled in some different way? What are the tree ranges in the FCS tree here? 

I'd expect these two snippets to have similar tree ranges in the FCS tree:
```fsharp
do
    let a = 1
    and b = 2

async {
    let! a = 1
    and! b = 2
}
```

If they're different only in the keyword range, then it'll likely be better to fix it in FCS instead of workarounding here.

Reply:
Simple and tokens are already processed in ProcessLocalBinding which targets SynExpr.Let and checks for isSecondary

ProcessAndLocalBinding targets specifically SynExpr.LetOrUse thus it looks only for AND_BANG token

Do you want to merge them together?

Reply:
Basically I only altered what already was here. Should we go for a new solution?

Reply:
OK, I see, it's not used for non-bang versions of the bindings. I'm OK with the change then. Though, if there's an easy way to unify handling for the bindings, I'd welcome such a change in a separate PR. 🙂